### ISSUE TITLE
Allow registering the same event in plugins multiple times

### ISSUE THREAD

**Issue by wkillerud** [cypress-io/cypress#19809]
### What would you like?

I'd like to be able to register functions for an event (`on("before:browser:launch", ...`) several times. Currently, every time an event is registered, [any existing callback is replaced](https://github.com/cypress-io/cypress/blob/5caf747573321003b8778aa28ceec608c75f54c0/packages/server/lib/plugins/index.js#L28).

I'm a bit unsure whether this counts as a feature request or a bug report honestly, but I figured this was never really intended and so is a new feature.

### Why is this needed?

I use a third party plugin (https://github.com/meinaart/cypress-plugin-snapshots) that [registers `before:browser:launch`](https://github.com/meinaart/cypress-plugin-snapshots/blob/6b95c7a930364b99f2496922159a6d693ad684a2/plugin.js#L22). At the same time, I would like to configure the browser to have `prefers-reduced-motion` turned on.

My plugin file ended up looking like this (edited for brevity):

```js
const { initPlugin } = require("cypress-plugin-snapshots/plugin");

module.exports = (on, config) => {
    on("before:browser:launch", (browser = {}, launchOptions) => {
        if (browser.family === "chromium" && browser.name !== "electron") {
            launchOptions.args.push("--force-prefers-reduced-motion");
        }
        return launchOptions;
    });

    initPlugin(on, config);
    return config;
};
```

My browser never got the argument, and `console.log` showed that the function never ran. I spent a while trying to debug before I thought of checking the plugin and Cypress source code to see if I might have some kind of collision happening, which turned out to be the case.

### Other

I made a minimal repo that demos the issue: https://github.com/wkillerud/cypress-multiple-register
When running `sample_spec.js`, it would be nice to have [both the registered functions](https://github.com/wkillerud/cypress-multiple-register/blob/main/cypress/plugins/index.js) run.

**Comment by hansireit** [comment]
I faced the same issue with the **cypress-plugin-snapshots** plugin, for now I got it working with the following workaround.

Create your own **initPlugin** function without the event registration. I know this is not possible for every plugin but for **cypress-plugin-snapshots** it works.

`
const { initConfig, CONFIG_KEY } = require('cypress-plugin-snapshots/src/config');
const initServer = require('cypress-plugin-snapshots/src/server/initServer');
const tasks = require('cypress-plugin-snapshots/src/tasks');

const initScreenshotPlugin = (on, globalConfig = {}) => {
  const config = initConfig(globalConfig.env[CONFIG_KEY]);
  initServer(config);

  globalConfig.env[CONFIG_KEY] = JSON.stringify(config);

  // Here I removed the "on('before:browser:launch..."

  on('task', tasks);
}

module.exports = {
  initScreenshotPlugin
};
`

Now you can use your custom function for initializing the plugin.
`
const { initScreenshotPlugin } = require("my-plugin-workaround");

module.exports = (on, config) => {
    on("before:browser:launch", (browser = {}, launchOptions) => {
        // ... 
    });

    initScreenshotPlugin(on, config);
    return config;
};
`

**Comment by romankhomitskyi** [comment]
Same here for cypress allure plugin

**Comment by TimVee** [comment]
Ran into this issue since I needed to set --window-size for the new chromium headless mode since cypress 12.15.

**Comment by svetlana-napris-elucid** [comment]
Same for me. I'm using **cypress-image-diff-js** plugin and need to set --window-size. 
My Cypress version v13.4.0 (browser - chrome, headless mode). 
The code I'm trying to use in my cypress.config.js copied from here: https://www.cypress.io/blog/2021/03/01/generate-high-resolution-videos-and-screenshots#the-window-size-on-ci

**Comment by claudio-bizzotto-zupit** [comment]
after 2 years there is still no idea on when this will be implemented. Everybody trying visual regression is facing this issue 😔

**Comment by abhishek-lambda** [comment]
yes in 2025 we are still facing this issue, @wkillerud can you please let us know if this will be ever picked, not being able to use multiple ` (on("before:browser:launch", ...) ` is breaking our tests at this point , pls check and help !

**Comment by abhishek-lambda** [comment]
@jennifer-shehane pls get this picked again ^
